src/etl/impl/LedgerLoader.hpp

```diff
@@ -136,19 +144,25 @@ class LedgerLoader {
             );
         }
 
-        // Remove all but the last NFTsData for each id. unique removes all but the first of a group, so we want to
-        // reverse sort by transaction index
-        std::sort(result.nfTokensData.begin(), result.nfTokensData.end(), [](NFTsData const& a, NFTsData const& b) {
-            return a.tokenID > b.tokenID && a.transactionIndex > b.transactionIndex;
-        });
+        auto const uniqueNftData = [](std::vector<NFTsData>& nfTokensData) {
+            // Remove all but the last NFTsData for each id. unique removes all but the first of a group, so we want
+            // to reverse sort by transaction index
+            std::sort(nfTokensData.begin(), nfTokensData.end(), [](NFTsData const& a, NFTsData const& b) {
+                return a.tokenID > b.tokenID && a.transactionIndex > b.transactionIndex;
+            });
+
+            // Now we can unique the NFTs by tokenID.
+            auto last = std::unique(nfTokensData.begin(), nfTokensData.end(), [](NFTsData const& a, NFTsData const& b) {
+                return a.tokenID == b.tokenID;
+            });
+            nfTokensData.erase(last, nfTokensData.end());
+        };
+
+        uniqueNftData(result.nfTokensData);
```

Comment:
Can any of these be empty? if yes, should we short circuit in the lambda?

Reply:
Yes. They could be empty 👍